### PR TITLE
deps: Include script to auto-build bottles

### PR DESCRIPTION
**Pull request by theopolis**
"Bottles" in Homebrew-terminology is the distributable version of a dependency. We build and store bottles from our dependencies to improve build time and to support more than just Ubuntu16.04 and MacOS10.12 for build systems.

Building these is difficult so we include a script to generate new bottles when a dependency formula is updated. Run `./tools/provision/deps_release.sh dependency1,dependency2` Where the comma-separated list is the set of dependencies changed. Each of these will be rebuilt. At the end the bottle `.tar.gz`s will be in the osquery repo root, ready to be uploaded to your hosting, and the formula will be updated to include the new bottle hashes.

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit f5aea56fc0025cd29924af6efea94112d1031476 (Job results: [3170](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3170/)) failed one or more tests (**Windows**).

File: tools/provision/darwin.sh [hunk 1]
```diff
@@ -7,7 +7,24 @@
...
+function vagrant_setup() {
+  sudo bash -c "$DARWIN_SETUP"
+}
+
 function distro_main() {
   GEM=`which gem`
   do_sudo $GEM install --no-ri --no-rdoc fpm
 }
+
+[ "$0" = "$BASH_SOURCE" ] && vagrant_setup
```
Comment (theopolis) [inline tools/provision/darwin.sh]: @javuto can you spot-check me here?

File: tools/provision/darwin.sh [hunk 2]
```diff
@@ -7,7 +7,24 @@
 #  LICENSE file in the root directory of this source tree. An additional grant
 #  of patent rights can be found in the PATENTS file in the same directory.
 
+DARWIN_SETUP="\
+if [[ ! -f /var/.osquery_build ]]; then \
+touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress; \
+PROD=\$(softwareupdate -l | grep \"\\*.*Command Line\" | \
```
Comment (javuto) [inline tools/provision/darwin.sh]: Will `softwareupdate` be always in `$PATH`?

File: tools/release/deps_release.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+function main() {
+  if [[ $# < 1 ]]; then
+    usage
+    exit 1
+  fi
+
+  CURRENT_DIR=$(pwd)
+  DEPS=$1
+
+  DEPS_CMD="cd /vagrant; make deps || true"
```
Comment (javuto) [inline tools/release/deps_release.sh]: Maybe put `/vagrant` in a variable for easier manipulation in the future

File: tools/release/deps_release.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  OSQUERY_BUILD_CPUS=4 vagrant up $LINUX_BOX
+  echo "[+] Building linux deps..."
+  vagrant ssh $LINUX_BOX -c "$DEPS_CMD"
+  install_deps $LINUX_BOX "$DEPS"
+  vagrant halt $LINUX_BOX
+
+  echo "[+] Vagrant up $DARWIN_BOX"
+  OSQUERY_BUILD_CPUS=4 vagrant up $DARWIN_BOX
+  echo "[+] Running initial softwareupdate check..."
+  vagrant ssh $DARWIN_BOX -c "/vagrant/tools/provision/darwin.sh"
```
Comment (javuto) [inline tools/release/deps_release.sh]: Use the `/vagrant` variable here

**Comment by osqueryer** [comment]
:-1: The commit 28a264a3c2b98e4676733e1b7815dcc5880b2cc3 (Job results: [2384](https://jenkins.osquery.io/job/osqueryPullRequestOSX/2384/)) failed one or more tests (**macOS/OS X**).

**Pull request merged**